packages/fxa-auth-server/fxa-oauth-server/lib/jwt_access_token.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const config = require('./config');
+const hex = require('buf').to.hex;
+const AppError = require('./error');
+const jwt = require('./jwt');
+
+const ISSUER = config.get('openid.issuer');
+
+const HEADER_TYP = 'at+JWT';
+
+/**
+ * Create a JWT access token from `grant`
+ */
+exports.create = async function generateJWTAccessToken(db, grant) {
+  const accessToken = await db.generateAccessToken(grant);
+  const clientId = hex(grant.clientId);
+
+  // Claims list from:
+  // https://tools.ietf.org/html/draft-bertocci-oauth-access-token-jwt-00#section-2.2
+  const claims = {
+    // The IETF spec for `aud` refers to https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+    // > REQUIRED. Audience(s) that this ID Token is intended for. It MUST contain the
+    // > OAuth 2.0 client_id of the Relying Party as an audience value. It MAY also contain
+    // > identifiers for other audiences. In the general case, the aud value is an array of
+    // > case sensitive strings. In the common special case when there is one audience, the
+    // > aud value MAY be a single case sensitive string.
+
+    // We only specify the clientId because we do not yet accept
+    // a `resource` parameter which we could use as the audience.
+    aud: [clientId],
+    client_id: clientId,
+    exp: Math.floor(accessToken.expiresAt / 1000),
+    iat: Math.floor(Date.now() / 1000),
+    iss: ISSUER,
+    jti: hex(accessToken.token),
+    scope: grant.scope.toString(),
+    sub: hex(grant.userId),
+  };
+
+  return {
+    ...accessToken,
+    jwt_token: await exports.sign(claims),
+  };
+};
+
+/**
+ * Sign a set of claims to create a JWT access token
+ *
+ * @param {Object} claims
+ * @returns {Promise<JWT>}
+ */
+exports.sign = function sign(claims) {
+  return jwt.sign(
+    {
+      // force an issuer to be set for direct calls to .sign,
+      // it can be overridden in the passed in claims.
+      iss: ISSUER,
```

Comment:
It's not a big deal, but I'd be tempted to move the setting (and checking) of `iss` up into `jwt.js`.  The issuer string is tightly bound to the signing keys so it would be conceptually cleaner to have them all specified at the same level of the stack here.  Anyone who wanted to set a custom issue would almost certainly want to also specify a different signing key.